app/cdap/components/GridTable/components/NestedMenu/index.tsx

```diff
@@ -31,34 +31,36 @@ export default function({
   setAnchorEl,
   open,
   handleMenuOpenClose,
-}: INestedMenuProps) {
-  const [anchorEl2, setAnchorEl2] = useState<EventTarget | null>(null);
-  const [nestedOptions, setNestedOptions] = useState<IMenuItem[]>([]);
+  newIndex,
+}: INestedMenuProps) => {
+  const [menuComponentOptions, setMenuComponentOptions] = useState([]);
   const classes = useNestedMenuStyles();
 
   const handleMenuClick = (
     event: React.MouseEvent<HTMLLIElement, MouseEvent>,
     menuItem: IMenuItem
   ) => {
-    setNestedOptions([]);
     event.preventDefault();
     event.stopPropagation();
-    if (menuItem?.options) {
-      setNestedOptions(menuItem.options);
-      setAnchorEl2(event.currentTarget);
+    if (menuItem?.options?.length > 0) {
+      setAnchorEl((prev) => [...prev, event.currentTarget]);
+      setMenuComponentOptions((prev) =>
+        prev.length ? [...prev, menuItem?.options] : [menuItem?.options]
+      );
     } else {
       submitMenuOption(menuItem.value, menuItem.supported_dataType);
       setAnchorEl(null);
       handleMenuOpenClose(title);
     }
   };
+  console.log('open 1', open);
   return (
     <>
       <Menu
         id="parent-menu"
         keepMounted
-        anchorEl={anchorEl}
-        open={open}
+        anchorEl={anchorEl?.length ? anchorEl[0] : null}
```

Comment:
we shouldnt depend on hardcoded index
this has to be completely dynamic

Reply:
This is for the one when we click on icon and menu open it will always be fixed to zero, for rest nested options index is managed dynamically